stix_shifter_modules/vectra/stix_transmission/connector.py

```diff
@@ -0,0 +1,215 @@
+        except Exception as ex:
+            if "timeout_error" in str(ex):
+                response_dict['code'] = 408
+            response_dict['message'] = str(ex)
+            self.logger.error('error when getting search results: %s', str(ex))
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+        return return_obj
+
+    async def create_results_connection(self, query, offset, length):
+        """
+         Function to create query connection
+        :param query: str
+        :param offset: int
+        :param length: str
+        :return: list
+        """
+        response_dict = {}
+        return_obj = {}
+        data = []
+        try:
+            offset = int(offset)
+            length = int(length)
+            pagination_offset = offset
+            total_records = length
+            page = 1
+
+            # changing offset to page number
+            if offset > length:
+                pagination_offset = offset % length
+                if pagination_offset < 5:
+                    page = round(offset / length) + 1
+                else:
+                    page = round(offset / length)
+
+            # set total record count
+            if self.api_client.result_limit < total_records:
+                total_records = self.api_client.result_limit
+
+            # set page size
+            if total_records <= Connector.VECTRA_MAX_PAGE_SIZE:
+                page_size = total_records
+            else:
+                page_size = Connector.VECTRA_MAX_PAGE_SIZE
+
+            query = f"page_size={page_size}&{query}&page={page}"
+            response_wrapper = await self.api_client.get_search_results(query)
+            response_code = response_wrapper.code
+            response = json.loads(response_wrapper.read().decode('utf-8'))
+
+            if response_code == 200:
+                data += response['results']
+                next_url = response.get('next')
+                while next_url:
+                    if len(data) >= total_records:
+                        break
+                    next_url = next_url.split('/api/v2.4/search/detections/?')
+                    query = next_url[1]
+                    next_response_wrapper = await self.api_client.get_search_results(query)
+                    response_code = next_response_wrapper.code
+                    next_response = json.loads(next_response_wrapper.read().decode('utf-8'))
+                    next_url = next_response.get('next')
+                    if response_code == 200:
+                        data += next_response['results']
+                    else:
+                        response = next_response
+                        break
+
+                if response_code == 200:
+                    return_obj['success'] = True
+                    return_obj['data'] = self.get_results_data(data[pagination_offset:total_records])
```

Comment:
why do you need to slice of the results list again since you are already checking the results count inside while loop?

Reply:
Please check the above comment; it explains slicing.